### ISSUE TITLE
Has anyone successfully trained on own dataset?

### ISSUE THREAD

**Issue by sjtuzyz** [foolwood/SiamMask#152]
Has anyone successfully trained on own dataset?Or I can only test it but cannot train?

**Comment by pvtien96** [comment]
me too

**Comment by JosieHong** [comment]
Hi, I have trained and tested it on my own dataset. Actually, I used SegTrack v2. 
And I put my works here: https://github.com/JosieHong/SiamMask_on_Your_Own_Dataset

I hope it is helpful.

**Comment by devendraswamy** [comment]
> Hi, I have trained and tested it on my own dataset. Actually, I used SegTrack v2.
> And I put my works here: https://github.com/JosieHong/SiamMask_on_Your_Own_Dataset
> 
> I hope it is helpful.

could you please help me to create own dataset ?

**Comment by QIQIVTrackingY** [comment]
hello! I want to on my own dataset. But in ./siammask_sharp/custom
if test:
            p0 = torch.nn.functional.pad(f[0], [16, 16, 16, 16])[:, :, 4*pos[0]:4*pos[0]+61, 4*pos[1]:4*pos[1]+61]
            p1 = torch.nn.functional.pad(f[1], [8, 8, 8, 8])[:, :, 2 * pos[0]:2 * pos[0] + 31, 2 * pos[1]:2 * pos[1] + 31]
            p2 = torch.nn.functional.pad(f[2], [4, 4, 4, 4])[:, :, pos[0]:pos[0] + 15, pos[1]:pos[1] + 15]
        else:
            p0 = F.unfold(f[0], (61, 61), padding=0, stride=4).permute(0, 2, 1).contiguous().view(-1, 64, 61, 61)
            if not (pos is None): p0 = torch.index_select(p0, 0, pos)
            p1 = F.unfold(f[1], (31, 31), padding=0, stride=2).permute(0, 2, 1).contiguous().view(-1, 256, 31, 31)
            if not (pos is None): p1 = torch.index_select(p1, 0, pos)
            p2 = F.unfold(f[2], (15, 15), padding=0, stride=1).permute(0, 2, 1).contiguous().view(-1, 512, 15, 15)
            if not (pos is None): p2 = torch.index_select(p2, 0, pos)

        if not(pos is None):
            p3 = corr_feature[:, :, pos[0], pos[1]].view(-1, 256, 1, 1)
        else:
            p3 = corr_feature.permute(0, 2, 3, 1).contiguous().view(-1, 256, 1, 1)

        out = self.deconv(p3)
        out = self.post0(F.upsample(self.h2(out) + self.v2(p2), size=(31, 31)))
        out = self.post1(F.upsample(self.h1(out) + self.v1(p1), size=(61, 61)))
        out = self.post2(F.upsample(self.h0(out) + self.v0(p0), size=(127, 127)))
        out = out.view(-1, 127*127)
        return out
p0,p1,p2 are not match p3. The outsize is 25*25 for p3 and is 17*17 for p0,p1,p2
